### PR TITLE
Docs: Add logo back to sidebar

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?
re #5500 

# Rationale for this change

While working https://github.com/apache/arrow-datafusion/pull/5670 we lost the sidebar logo

Screenshot 2023-03-22 at 7 22 38 AM

# What changes are included in this PR?

Add the logo back in

Screenshot 2023-03-22 at 7 21 55 AM

# Are these changes tested?

Tested locally

# Are there any user-facing changes?

docs

### CODE REVIEW

File: docs/source/_templates/docs-sidebar.html [hunk 1]
```diff
@@ -14,3 +14,8 @@
     {% endif %}
   </div>
 </nav>
+
+
+<a class="navbar-brand" href="{{ pathto(master_doc) }}">
+  <img src="{{ pathto('_static/images/DataFusion-Logo-Background-White.png', 1) }}" class="logo" alt="logo">
```
Comment (alamb) [inline docs/source/_templates/docs-sidebar.html:20]: this used to be

```
  <img src="{{ pathto('_static/images' + logo, 1) }}" class="logo" alt="logo">
```

But then sphinx complained that `logo` wasn't defined -- so I just hard coded it to match the main logo

https://github.com/apache/arrow-datafusion/blob/af97ac886c425efefb0536c5344894703f65d7fa/docs/source/conf.py#L103

**Review by jackwener — APPROVED** [review]

**Comment by alamb** [comment]
It is so cool it is already live on https://arrow.apache.org/datafusion/ ❤️ 

Screenshot 2023-03-22 at 9 58 03 AM

**Comment by alamb** [comment]
>  Kinda stupid question... why logo is at the bottom? 🤔


TLDR is "I don't know"  -- I was just putting it back where I found it
